woocommerce/google-listings-and-ads
# Tours API Endpoint

### Changes proposed in this Pull Request:

This PR implements an API endpoint for handling the Tour Visualisation.

Tours are going to be a new experience provided by [Woo Spotlight component.](https://ghidorahp2.wordpress.com/2022/04/13/shared-in-app-spotlight-component/) 

When the user is entering the first time, is going to see this tour spotlights. When the user clicks on "Got it" button. We won't show this anymore. So we will save this as an option in the Database. 

For now it will be only 1 tour for Attribute Mapping. But in next iteration we will show a new one in case the user has invalid product data. So, if in the future there are more tours. We can add them here as well.

### Detailed test instructions:

1.  Checkout this PR
2.  Create a new tour by requesting `POST wp-json/wc/gla/tours/` with this body:
```
{
    id: 'test',
    checked: true,
}
```
3. Get the tour by requesting `GET wp-json/wc/gla/tours/test`
4. You should get the tour you just created
5. Try to get a not existing tour, and try to send invalid data like missing props or props with wrong types.
